src/App.tsx

```diff
@@ -1,26 +1,169 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import { FC, useState, useEffect, useMemo, useCallback } from 'react';
+import {
+  USER_ID,
+  getTodos,
+  addTodo,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorMessage } from './types/ErrorMessage';
+import { StatusFilter } from './types/StatusFilter';
+import { TodosHeader } from './components/TodosHeader';
+import { TodoList } from './components/TodoList';
+import { TodosFooter } from './components/TodosFooter';
+import { ErrorNotification } from './components/ErrorNotification';
 
-const USER_ID = 0;
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage | null>(null);
+  const [filterValue, setFilterValue] = useState<StatusFilter>(
+    StatusFilter.All,
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todoIdsToDel, setTodoIdsToDel] = useState<number[]>([]);
+  const [todoIdsToUpdate, setTodoIdsToUpdate] = useState<number[]>([]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        setTodos(await getTodos());
+      } catch (err) {
+        setErrorMessage(ErrorMessage.LoadError);
+      }
+    })();
```

Comment:
1) will be better, right?
2) 2) It is better to move all `useEffect` to the bottom(before render), for better support, and readability
```suggestion
    const getAllTodos = async () => {
      try {
        setTodos(await getTodos());
      } catch (err) {
        setErrorMessage(ErrorMessage.LoadError);
      }
    }
    
    getAllTodos();
```